decompiler/backend/variabledeclarations.py

```diff
@@ -66,6 +73,16 @@ def _generate_definitions(global_variables: set[GlobalVariable]) -> Iterator[str
                     if not variable.type.type in [Integer.char(), Integer.uint8_t(), CustomType.wchar16(), CustomType.wchar32()]:
                         br, bl = "{", "}"
                     yield f"{base}{variable.type.type} {variable.name}[{hex(variable.type.elements)}] = {br}{CExpressionGenerator().visit(variable.initial_value)}{bl};"
+                case Struct():
+                    if is_struct_string(variable.type):
+                        yield base + f"struct {variable.type.name} {variable.name} = {CExpressionGenerator().visit(get_data_of_struct_string(variable))};"
+                        continue
+                    string = f"struct {variable.type.name} {variable.name}" + "{\n"
```

Comment:
This code snipped generates the code for the declaration and initialization of structs, which are global variables. There are two cases: the normal case (l. 80) and syntax sugar for struct based stings (l.77 - l.80). AFAIK we do not emit any typedefs, therefore the typename still needs to be prefaced with `struct`.

Reply:
Hm I think I need to clarify, sorry for the confusion.
A struct in C looks kinda like this: (brazenly stolen from https://www.w3schools.com/c/c_structs.php)
```C
struct MyStructure {   // Structure declaration
  int myNum;           // Member (int variable)
  char myLetter;       // Member (char variable)
}; // End the structure with a semicolon
```
But the code emits three words on the first line:
```python
string = f"struct {variable.type.name} {variable.name}" + "{\n"
```
So I was wonderding - if `{variable.type.name}` is  `MyStructure`, when what is `{variable.name}` about?
I now see that you actually wanted to do a inline declaration here in the form of this:
```C
struct myStructure s1 = {13, 'B', "Some text"};
```
It was just lost on me since the function emitting this string is actually called `_generate_definitions`. So it is supposed to be a declaration?

Reply:
Yes, exactly, it is supposed to be declaration (+ initialization). We use the "designated initializer" syntax, so that the reader does not need to memorize the struct definition to understand the initialization.
What do you think about renaming the method to `_generate_declarations_and_initializations`?

Reply:
I think it is fine, was just being confused